packages/fs-detectors/src/detect-framework.ts

```diff
@@ -90,3 +150,75 @@ export async function detectFramework({
   );
   return result.find(res => res !== null) ?? null;
 }
+
+/**
+ * Framework with a `detectedVersion` specifying the version
+ * or version range of the relevant package
+ */
+type VersionedFramework = Framework & {
+  detectedVersion?: string;
+};
+
+// Note: Does not currently support a `frameworkList` of monorepo managers
+export async function detectFrameworkRecord({
+  fs,
+  frameworkList,
+}: DetectFrameworkRecordOptions): Promise<VersionedFramework | null> {
+  const result = await Promise.all(
+    frameworkList.map(async frameworkMatch => {
+      const matchResult = await matches(fs, frameworkMatch);
+      if (matchResult) {
+        return {
+          ...frameworkMatch,
+          detectedVersion: matchResult?.detectedVersion,
+        };
+      }
+      return null;
+    })
+  );
+  const frameworkRecord = result.find(res => res !== null) ?? null;
+
+  return frameworkRecord;
+}
+
+export function detectFrameworkVersion(
+  frameworkRecord: Framework
+): string | undefined {
+  const allDetectors = [
+    ...(frameworkRecord.detectors?.every || []),
+    ...(frameworkRecord.detectors?.some || []),
+  ];
+  const firstMatchPackage = allDetectors.find(d => d.matchPackage);
+
+  if (!firstMatchPackage?.matchPackage) {
+    return;
+  }
+
+  const version = lookupInstalledVersion(
+    process.execPath,
+    firstMatchPackage.matchPackage
+  );
+  if (version) {
+    return version;
+  }
+
+  return;
```

Comment:
```suggestion
  const version = lookupInstalledVersion(
    process.execPath,
    firstMatchPackage.matchPackage
  );
  
  return version;
```

Suggestion: This should be fine unless `version` can be returned as some kinda falsy value and you want to explicitly return `undefined` _*cough* *cough* explicit conditionals *cough* *cough*_

Reply:
Done! [`91f922b` (#9009)](https://github.com/vercel/vercel/pull/9009/commits/91f922b1d03f6721da02529a71b473e1b6cb8d1e)